### ISSUE TITLE
NWN Enhanced Edition Extension/NWN 1

### ISSUE THREAD

**Issue by HadToRegister** [Nexus-Mods/Vortex#6250]
**Describe the bug**
Using this extension with NWNEE
The extension doesn't copy all of the files to the proper directories, and instead skips a lot of the files completely that are included in the zip, despite being in the proper folders inside the zip.

All the files mentioned below, are inside the appropriate folders inside the zip file, but are ignored.

WAV files don't get installed to the AMBIENT folder, and instead are skipped
CDX, DBF, FPT files don't get installed to the DATABASE folder, and are skipped
NBM, BIK files don't get installed to the MOVIES folder, and are skipped
2DA, UTI, TXI, MDL don't get installed to the OVERRIDE folder and are skipped
EXE, HIF, ERF files don't get installed to the HAK folder and are skipped (sometimes, ERF and EXE are packaged to use in the Hak folder)

**To Reproduce**
1. Download any mod for NWN1 or NWNEE that has any of these extensions set to install in any of the mentioned folders, and they'll be skipped by the Extension when installed.

The other method I did, was to subscribe to a mod in the Steam workshop, open up Steam\steamapps\workshop\content\704450\ open the folder and zip up any folders installed to the new folder in the root, such as 'override', 'ambient' etc, preserving the folder structure, just like any Nexus NWN mod.
The mods installed properly, except for ALL of the file extension I mentioned for the appropriate folders. Instead, those extensions, were all skipped.
(BTW, the Steam Workshop technique I used was suggested by people using NWNEEE and Beamdog)

**Expected behavior**
I expected Files with the wav, cdx, dbf, fpt, wbm, bik, 2da, uti, txi, mdl, exe, hif, erf to install to the folders that they were assigned when installing by hand or with steam workshop.
Instead they were all skipped.

**Platform (please complete the following information):**
 - OS: Windows
 - Vortex Version 1.1.15 I KNOW it's an old version, but the problem is with the Extension, and I couldn't find a place to post an issue in the Vortex - Games section.
Going through the changelogs, I haven't seen any changes to the NWN extension from 1.1.15 to 1.2.11

**Comment by IDCs** [comment]
Hi @HadToRegister, thanks for letting us know and providing target directories - the game extension uses file extensions to figure out where to place files and will ignore any unknown file extensions (we chose to go this route as many mod authors pack their files loosely at the archive's root)

I will enhance the game extension to support those file types.

**Comment by IDCs** [comment]
Hi @HadToRegister, I just read through our extension's code and we're currently deploying ERF files into the ERF folder - are you aware of any issues that may arise if I change that to use the HAK folder instead ?

Can you think of a way we could identify which ERF file goes into which folder?

**Comment by HadToRegister** [comment]
Sometimes a modder will put an ERF into the HAK folder, along with having ERF files in the ERF folder.

It's best to have the files go into the folders by what folder they were in, inside the zip file, otherwise if going by their extension, files will go everywhere.

So if a zip file has the folder structure

ERF folder sometimes has other file types in it besides ERF, they should go into the folder in the computer, that they originated from, from the zip file

HAK folder sometimes has other file types in it besides HAK, such as ERF occasionally, (along with EXE (exe for converting mods to work with a specific Mod etc), so any file that was in the HAK folder in the zip, should be sent to the HAK folder on the computer.

Basically a 1:1 copy of the file and folder structure of the zip

I honestly don't know why modders do this, but I see it a LOT while using mods, so the sheer number of these instances happening, (Example: ERF in a HAK folder etc) can't be 'Mistakes', they must be put there for a specific purpose.
So, perhaps a 1:1 copy of the Zip folder structure, rather than by file-type.

**Comment by HadToRegister** [comment]
For mods that don't have a directory structure, such as "Fast Sleep" that just has a *.2da file zipped up, then yes, that file needs to go to "override" (which by the way it didn't, it just went to the root directory of \Documents\Neverwinter Nights\

That would be a good reason for File Type installation, when the zip file has no directory structure, which most of the times just means a lone HAK or 2DA file.
Most Mods though, have a folder structure though, but something like PRC 3.8.3 which I installed using Steam Workshop, then zipping it up with the folder structure intact, had an ERF in the HAK folder, and this is version 3.8.3 so there must be a reason for it being there.
It might be for conversion purposes.

**Comment by HadToRegister** [comment]
OK, one more file file type that isn't getting copied/installed.

NCS should go in the override folder, however those are being skipped as well.
(I assume that NCS means "Neverwinter Compiled Script)

Try out the Inventory Sort Mod 2.0 from the Neverwinter Vault
https://neverwintervault.org/sites/neverwintervault.org/files/project/29335/files/inventory_sort_2.0.7z

**Comment by IDCs** [comment]
Hi @HadToRegister, I've modified our extension (https://github.com/Nexus-Mods/vortex-games/commit/c76b1a3acc473ec0e5056bb43163d3f7a24d071a) in the following manner: 

The extension will try to use the existing directory structure of a mod when a correct structure is provided (inventory_sort is a good example of such a mod). If the correct structure is _not_ provided, we will fallback to the custom installer which now has support for all the file extensions you mentioned.

Please keep in mind that when a mod is packed loosely (different file types at the root of the archive with no clear directory structure) .erf files will still be deployed to the ERF directory as there's no way for us to resolve the correct file destination.

**Comment by HadToRegister** [comment]
Sounds like a great solution, thanks!
